source/RoslynAnalyzers/MessageContractAnalyzers_Conventions.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using static Octopus.RoslynAnalyzers.Descriptors;
+
+namespace Octopus.RoslynAnalyzers
+{
+    // ReSharper disable UnusedMethodReturnValue.Local
+    
+    // This part of the MessageContractAnalyzers contains all the specific methods which enforce each convention
+    public partial class MessageContractAnalyzers
+    {
+       
+        static bool ApiContractTypes_MustLiveInTheAppropriateNamespace(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDec)
```

Comment:
> That could also be helpful and allow the descriptors to live in the same place as the analyzer?

The descriptors _can_ live in the same place as the analyzer. I just moved them because the class was too big 🤣 

> Consider a class per check - that could enable some funky stuff around "just add a file that implements this interface and it gets picked up automatically"...

I think perf considerations mean that the normal dynamic/discovery stuff we do in the server is off-limits here.

Reply:
Yeah, the perf trade offs here are much stronger than normal, so yeah, dynamic stuff is probably out of scope.
Then again, maybe it's just something we do once and cache.

I've probably got an overdeveloped "keep things small" instinct. :)